dashboard/app/controllers/courses_controller.rb

```diff
@@ -189,6 +143,53 @@ def i18n_params
 
   private
 
+  def get_unit_group
+    course_name = params[:course_name]
+
+    unit_group =
+      params[:action] == "edit" ?
+        UnitGroup.get_without_cache(course_name) :
+        UnitGroup.get_from_cache(course_name)
+
+    return unit_group if unit_group
+
+    # When the url of a course family is requested, redirect to a specific course version.
+    if UnitGroup.family_names.include?(params[:course_name])
+      unit_group = UnitGroup.latest_stable_version(params[:course_name])
+      redirect_to action: params[:action], course_name: unit_group.name if unit_group
+    end
+
+    unit_group
+  end
+
+  def set_unit_group
+    @unit_group = get_unit_group
+    raise ActiveRecord::RecordNotFound unless @unit_group
+  end
+
+  def render_no_access
+    if params[:action] == "show" && @unit_group.plc_course
+      authorize! :show, Plc::UserCourseEnrollment
+      user_course_enrollments = [Plc::UserCourseEnrollment.find_by(user: current_user, plc_course: @unit_group.plc_course)]
+      render 'plc/user_course_enrollments/index', locals: {user_course_enrollments: user_course_enrollments}
+      return
+    end
```

Comment:
would you consider extracting this section into its own method with a name that more accurately describes what it's doing? I think it is a bit more readable to be able to call this as something like `before_action :check_plc_enrollment, only: [:show]` rather than embedding it in this other method.

Reply:
yes will do